.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedDateTimeParser.cs

```diff
@@ -143,6 +143,11 @@ public static void AddPeriodToResolution(Dictionary<string, string> resolutionDi
 
             if (!string.IsNullOrEmpty(start) && !string.IsNullOrEmpty(end))
             {
+                if (start.StartsWith(DateMinString) || end.StartsWith(DateMinString))

```

Comment:
Can you make this into a method? Call it "checkUnresolvedDates".
We should also probably mark the res parameter passed to this method as 'ref' to highlight it will be changed.